processing/src/main/java/org/apache/druid/query/metadata/SegmentAnalyzer.java

```diff
@@ -198,26 +199,42 @@ private ColumnAnalysis analyzeStringColumn(
 
     Comparable min = null;
     Comparable max = null;
-
-    if (!capabilities.hasBitmapIndexes()) {
-      return ColumnAnalysis.error("string_no_bitmap");
-    }
-
-    final BitmapIndex bitmapIndex = columnHolder.getBitmapIndex();
-    final int cardinality = bitmapIndex.getCardinality();
-
-    if (analyzingSize()) {
-      for (int i = 0; i < cardinality; ++i) {
-        String value = bitmapIndex.getValue(i);
-        if (value != null) {
-          size += StringUtils.estimatedBinaryLengthAsUTF8(value) * bitmapIndex.getBitmap(bitmapIndex.getIndex(value)).size();
+    int cardinality = 0;
+    if (capabilities.hasBitmapIndexes()) {
+      final BitmapIndex bitmapIndex = columnHolder.getBitmapIndex();
+      cardinality = bitmapIndex.getCardinality();
+
+      if (analyzingSize()) {
+        for (int i = 0; i < cardinality; ++i) {
+          String value = bitmapIndex.getValue(i);
+          if (value != null) {
+            size += StringUtils.estimatedBinaryLengthAsUTF8(value) * bitmapIndex.getBitmap(bitmapIndex.getIndex(value))
+                                                                                .size();
+          }
         }
       }
-    }
 
-    if (analyzingMinMax() && cardinality > 0) {
-      min = NullHandling.nullToEmptyIfNeeded(bitmapIndex.getValue(0));
-      max = NullHandling.nullToEmptyIfNeeded(bitmapIndex.getValue(cardinality - 1));
+      if (analyzingMinMax() && cardinality > 0) {
+        min = NullHandling.nullToEmptyIfNeeded(bitmapIndex.getValue(0));
+        max = NullHandling.nullToEmptyIfNeeded(bitmapIndex.getValue(cardinality - 1));
+      }
+    } else if (capabilities.isDictionaryEncoded()) {
+      // fallback if no bitmap index
```

Comment:
I dunno, maybe we should just remove the `"size"` analysisType. It is nothing but trouble.

Reply:
Agree. I think it would be interesting to have size be representative of "estimated size in memory" for incremental index, or size in bytes of column in the segment file that is mapped, but what it's computing here seems like basically nonsense for all column types.

Reply:
> or size in bytes of column in the segment file that is mapped

`dictionary size + bitmap size` I vote for that too.

Reply:
Raised proposal #7124

Reply:
Until #7124 is sorted, what you're doing here (setting size to 0 when there's no bitmap index) sounds good to me.